beetsplug/parentwork.py

```diff
@@ -0,0 +1,142 @@
+# -*- coding: utf-8 -*-
+# This file is part of beets.
+# Copyright 2017, Dorian Soergel.
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+
+"""Gets work title, disambiguation, parent work and its disambiguation, composer, composer sort name and performers
+"""
+
+from __future__ import division, absolute_import, print_function
+
+import subprocess
+
+from beets import ui
+from beets import util
+from beets.plugins import BeetsPlugin
+
+import musicbrainzngs
+
+class ParentWorkPlugin(BeetsPlugin):
+
+    def __init__(self):
+        super(ParentWorkPlugin, self).__init__()
+        self.import_stages = [self.imported]
+    # possible options
+    #        self.config.add({
+    #            u'bin': u'KeyFinder',
+    #            u'auto': True,
+    #            u'overwrite': False,
+    #        })
+    #        self.add_media_field('work', 'work title')
+    #        self.add_media_field('work_disamb', 'work title disambiguation')
+    #        self.add_media_field('parent_work', 'parent work title')
+    #        self.add_media_field('parent_work_disamb', 'parent work title disambiguation')
+    #        self.add_media_field('parent_composer', 'parent composer name')
+    #        self.add_media_field('parent_composer_sort', 'parent composer sort name')
+    #        self.add_media_field('performer', 'performer name')
+    #        self.add_media_field('performer_sort', 'performer sort name')
+    
+    def commands(self):
+        cmd = ui.Subcommand('parentwork',
+                            help=u'fetches parent works, composers and performers')
+        cmd.func = self.command
+        return [cmd]
+        
+    def command(self, lib, opts, args):
+        self.find_key(lib.items(ui.decargs(args)))
+             
+    def imported(self, session, task):
+        self.find_work(task.items)        
+    
+    def find_work(self, items):
+        
+        for item in items:
+            performer            = []
+            performer_sort       = []
+            work                 = []
+            work_disambig        = []
+            parent_work          = []
+            parent_work_disambig = []
+            parent_composer      = []
+            parent_composer_sort = []
+            item.read()
+            recording_id=item['mb_trackid']
+            i=0
+            while i<5:
```

Comment:
I get an error 503 from the musicbrainz server quite regularly. To avoid this I do several tries. However, maybe it would be better to try the individual queries multiple times rather than the whole set